### DISCUSSION TITLE
`noConstantCondition` and `while(true) {}`

### DISCUSSION THREAD

**Discussion by Conaclos** [rome/tools#4746] [Suggestions]
Using the constant condition `true` in a while loop seems a common idiom in JavaSCript:

```js
while (true) {
  if (someCondition) {
    break;
  }
}
```

On the Typescript Compiler code-base it appears 94 times.

I think we could relax `noConstantCondition` to allow this idiom.

Opinions?

**Comment by ematipico** [comment]
How do you propose to relax the condition? An option? Or with a specific check?

**Reply by Conaclos** [reply]
Just a specific check that allows `true` constant condition in `while`.
This is a very simple change.

This could diverge from the _ESlint_ implementation.
However, this seems unlikely that `while (true)` is a mistake. This looks like a deliberate choice.

**Reply by ematipico** [reply]
I don't mind diverging from ESLint, Rome is not meant to be a 1:1 copy of ESLint.

**Reply by Conaclos** [reply]
Let's just think some days about that before proposing a PR.
